### ISSUE TITLE
Missing GCP regions in catalog

### ISSUE THREAD

**Issue by bilelomrani1** [skypilot-org/skypilot#1832]
Currently the GCP catalog contains instances in the following regions:
```
List of supported gcp regions: 'asia-east1, europe-west4, us-central1, us-central2, us-east1, us-east4, us-east5, us-south1, us-west1, us-west2, us-west3, us-west4'
```
Is there a reason why many regions in Europe and Asia are missing (for instance `europe-west1`, `europe-west2`...)? The [Gcloud documentation](https://cloud.google.com/compute/docs/regions-zones) lists all available regions and many of them are missing. I am working with a Google Bucket located in Europe and would like to provision machines in the same zones to limit egress costs.

**Comment by concretevitamin** [comment]
Our [GCP catalog fetcher](https://github.com/skypilot-org/skypilot/blob/master/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py) seems to be missing some regions.  There's no particular reason not to fix this except for potential parsing issues.

cc @WoosukKwon @infwinston who worked on a better fetcher: https://github.com/skypilot-org/skypilot/pull/1642. However, using this branch and re-fetching, I do not see `europe-west1` being pulled down.

**Comment by concretevitamin** [comment]
@bilelomrani1 If this is a blocker, here's one way to unblock you by using the new fetcher PR and https://skypilot.readthedocs.io/en/latest/reference/faq.html#advanced-how-to-make-skypilot-use-all-global-regions:

```bash
# Assuming you have installed SkyPilot from source.
cd <skypilot repo>
gh pr checkout 1642

cd ~/.sky/catalogs/v5
mv gcp gcp.backup
python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions
```
This takes a minute or so and now I see `europe-west{1,2}` etc.

**Comment by flowinh2o** [comment]
> @bilelomrani1 If this is a blocker, here's one way to unblock you by using the new fetcher PR and https://skypilot.readthedocs.io/en/latest/reference/faq.html#advanced-how-to-make-skypilot-use-all-global-regions:
> 
> ```shell
> # Assuming you have installed SkyPilot from source.
> cd <skypilot repo>
> gh pr checkout 1642
> 
> cd ~/.sky/catalogs/v5
> mv gcp gcp.backup
> python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions
> ```
> 
> This takes a minute or so and now I see `europe-west{1,2}` etc.

For some reason I could not get this to work on my mac m1 and tried the following:
```
cd <skypilot repo>
git checkout origin/reimplement-gcp-fetcher
pip install ".[gcp,aws]"
cd ~/.sky/catalogs/v5
mv gcp gcp.backup
python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions
```

```
> sky launch -c gcp-test gcp-test.yaml
Task from YAML spec: gcp-test.yaml
ValueError: Accelerator "A100-80GB" is not available in "asia-southeast1". ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- sky py
> sky show-gpus --cloud gcp --region asia-southeast
# no output
```

**Comment by flowinh2o** [comment]
> @flowinh2o Ah, the new fetcher is a more general fix than needed if you're just looking to get `A100-80GB` in that new region.
> 
> I've tried the following and it works for me:
> 
> ```
> git checkout master
> cd ~/.sky/catalogs/v5
> mv gcp gcp.backup
> sky launch --gpus A100-80GB:8 --region asia-southeast1 --cloud gcp         
> ```
> 
> i.e., no need to switch to the new fetcher branch and run it.
Looks like I had to do a "pip uninstall skypilot" and then "pip install skypilot" without using the gcp,aws option and things are working now.  Thanks.

**Comment by flowinh2o** [comment]
> @flowinh2o Ah, the new fetcher is a more general fix than needed if you're just looking to get `A100-80GB` in that new region.
> 
> I've tried the following and it works for me:
> 
> ```
> git checkout master
> cd ~/.sky/catalogs/v5
> mv gcp gcp.backup
> sky launch --gpus A100-80GB:8 --region asia-southeast1 --cloud gcp         
> ```
> 
> i.e., no need to switch to the new fetcher branch and run it.
Looks like I had to do a "pip uninstall skypilot" and then "pip install skypilot" without using the gcp,aws option and things are working now.  Thanks.

**Comment by bilelomrani1** [comment]
I @concretevitamin, thank you for your help, I tried to follow the steps you provided in the FAQ, but unfortunately there seems to be a parsing error somewhere which prevents the catalog to be created:

```shell
$ python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions

Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 588, in <module>
    gcp_catalog_df = get_catalog_df(region_prefix_filter)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 560, in get_catalog_df
    gpu_df = get_gpu_df(region_prefix)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 425, in get_gpu_df
    gpu_pricing = get_gpu_price_table(gpu_price_table_url[0])
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 375, in get_gpu_price_table
    df['AcceleratorCount'] = df['AcceleratorName'].apply(
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/series.py", line 4626, in apply
    return SeriesApply(self, func, convert_dtype, args, kwargs).apply()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/apply.py", line 1025, in apply
    return self.apply_standard()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/apply.py", line 1076, in apply_standard
    mapped = lib.map_infer(
  File "pandas/_libs/lib.pyx", line 2834, in pandas._libs.lib.map_infer
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 376, in <lambda>
    lambda x: GPU_TYPES_TO_COUNTS[x])
KeyError: 'NVIDIA\n            L4'
```

**Comment by Michaelvll** [comment]
> Hi @concretevitamin, thank you for your help, I tried to follow the steps you provided in the FAQ, but unfortunately there seems to be a parsing error somewhere which prevents the catalog to be created:
> 
> ```shell
> $ python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions
> 
> Traceback (most recent call last):
>   File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 588, in <module>
>     gcp_catalog_df = get_catalog_df(region_prefix_filter)
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 560, in get_catalog_df
>     gpu_df = get_gpu_df(region_prefix)
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 425, in get_gpu_df
>     gpu_pricing = get_gpu_price_table(gpu_price_table_url[0])
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 375, in get_gpu_price_table
>     df['AcceleratorCount'] = df['AcceleratorName'].apply(
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/series.py", line 4626, in apply
>     return SeriesApply(self, func, convert_dtype, args, kwargs).apply()
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/apply.py", line 1025, in apply
>     return self.apply_standard()
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/apply.py", line 1076, in apply_standard
>     mapped = lib.map_infer(
>   File "pandas/_libs/lib.pyx", line 2834, in pandas._libs.lib.map_infer
>   File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 376, in <lambda>
>     lambda x: GPU_TYPES_TO_COUNTS[x])
> KeyError: 'NVIDIA\n            L4'
> ```

Hi @bilelomrani1, thank you for trying that out! Are you currently on the latest master branch? We recently fixed the fetcher for GCP and merged into the master branch #1642. Do you mind trying it again?

We are planning to have the resources from all the regions in the GCP catalog hosted by SkyPilot, so the users do not need to run this script themselves.

**Comment by bilelomrani1** [comment]
Hosting the catalog yourself would be a great idea! On master, I get another exception, I provided the stacktrace below. 

<details close>
<summary>Stacktrace</summary>
<br>
 python -m sky.clouds.service_catalog.data_fetchers.fetch_gcp --all-regions

Fetching machine types for zone 'us-east1-b'...
Fetching machine types for zone 'us-east4-b'...
Fetching machine types for zone 'us-central1-f'...
Fetching machine types for zone 'us-west1-a'...
Fetching machine types for zone 'europe-west3-a'...
Fetching machine types for zone 'europe-west1-b'...
Fetching machine types for zone 'asia-southeast1-b'...
Fetching machine types for zone 'europe-west2-a'...
Fetching machine types for zone 'europe-west3-b'...
Fetching machine types for zone 'asia-east1-b'...
Fetching machine types for zone 'europe-west1-d'...
Fetching machine types for zone 'europe-west1-c'...
Fetching machine types for zone 'europe-west2-c'...
Fetching machine types for zone 'europe-west3-c'...
Fetching machine types for zone 'europe-west2-b'...
Fetching machine types for zone 'us-east4-a'...
Fetching machine types for zone 'asia-southeast1-a'...
Fetching machine types for zone 'asia-northeast1-c'...
Fetching machine types for zone 'us-east1-c'...
Fetching machine types for zone 'asia-south1-a'...
Fetching machine types for zone 'us-central1-b'...
Fetching machine types for zone 'us-central1-c'...
Fetching machine types for zone 'europe-west4-a'...
Fetching machine types for zone 'europe-west4-b'...
Fetching machine types for zone 'europe-west4-c'...
Fetching machine types for zone 'us-east1-d'...
Fetching machine types for zone 'southamerica-east1-b'...
Fetching machine types for zone 'us-central1-a'...
Fetching machine types for zone 'us-west1-b'...
Fetching machine types for zone 'asia-east1-a'...
Fetching machine types for zone 'us-east4-c'...
Fetching machine types for zone 'asia-northeast1-a'...
Fetching machine types for zone 'asia-east2-b'...
Fetching machine types for zone 'asia-northeast2-c'...
Fetching machine types for zone 'asia-southeast1-c'...
Fetching machine types for zone 'us-west1-c'...
Fetching machine types for zone 'asia-east2-c'...
Fetching machine types for zone 'asia-south2-a'...
Fetching machine types for zone 'asia-northeast1-b'...
Fetching machine types for zone 'southamerica-east1-c'...
Fetching machine types for zone 'australia-southeast1-b'...
Fetching machine types for zone 'asia-northeast2-a'...
Fetching machine types for zone 'southamerica-east1-a'...
Fetching machine types for zone 'asia-southeast2-b'...
Fetching machine types for zone 'asia-east1-c'...
Fetching machine types for zone 'asia-south1-c'...
Fetching machine types for zone 'asia-south2-b'...
Fetching machine types for zone 'asia-east2-a'...
Fetching machine types for zone 'asia-northeast3-a'...
Fetching machine types for zone 'australia-southeast2-c'...
Fetching machine types for zone 'asia-south1-b'...
Fetching machine types for zone 'australia-southeast1-c'...
Fetching machine types for zone 'asia-northeast2-b'...
Fetching machine types for zone 'asia-southeast2-c'...
Fetching machine types for zone 'asia-northeast3-b'...
Fetching machine types for zone 'europe-north1-a'...
Fetching machine types for zone 'europe-north1-b'...
Fetching machine types for zone 'europe-central2-a'...
Fetching machine types for zone 'asia-south2-c'...
Fetching machine types for zone 'europe-southwest1-b'...
Fetching machine types for zone 'europe-central2-b'...
Fetching machine types for zone 'europe-southwest1-c'...
Fetching machine types for zone 'europe-north1-c'...
Fetching machine types for zone 'australia-southeast1-a'...
Fetching machine types for zone 'europe-west12-c'...
Fetching machine types for zone 'europe-southwest1-a'...
Fetching machine types for zone 'europe-west12-a'...
Fetching machine types for zone 'europe-central2-c'...
Fetching machine types for zone 'europe-west6-a'...
Fetching machine types for zone 'europe-west8-a'...
Fetching machine types for zone 'asia-northeast3-c'...
Fetching machine types for zone 'australia-southeast2-a'...
Fetching machine types for zone 'europe-west12-b'...
Fetching machine types for zone 'europe-west6-b'...
Fetching machine types for zone 'europe-west9-b'...
Fetching machine types for zone 'me-central1-c'...
Fetching machine types for zone 'europe-west8-b'...
Fetching machine types for zone 'europe-west6-c'...
Fetching machine types for zone 'europe-west9-c'...
Fetching machine types for zone 'northamerica-northeast1-a'...
Fetching machine types for zone 'northamerica-northeast2-b'...
Fetching machine types for zone 'southamerica-west1-c'...
Fetching machine types for zone 'me-central1-a'...
Fetching machine types for zone 'europe-west8-c'...
Fetching machine types for zone 'europe-west9-a'...
Fetching machine types for zone 'northamerica-northeast1-b'...
Fetching machine types for zone 'me-west1-a'...
Fetching machine types for zone 'northamerica-northeast2-c'...
Fetching machine types for zone 'us-south1-a'...
Fetching machine types for zone 'asia-southeast2-a'...
Fetching machine types for zone 'me-central1-b'...
Fetching machine types for zone 'northamerica-northeast1-c'...
Fetching machine types for zone 'me-west1-b'...
Fetching machine types for zone 'southamerica-west1-a'...
Fetching machine types for zone 'us-south1-b'...
Fetching machine types for zone 'australia-southeast2-b'...
Fetching machine types for zone 'us-east5-a'...
Fetching machine types for zone 'us-west2-b'...
Fetching machine types for zone 'northamerica-northeast2-a'...
Fetching machine types for zone 'us-west2-c'...
Fetching machine types for zone 'us-west3-c'...
Fetching machine types for zone 'me-west1-c'...
Fetching machine types for zone 'us-south1-c'...
Fetching machine types for zone 'us-east5-b'...
Fetching machine types for zone 'us-west3-a'...
Fetching machine types for zone 'us-west2-a'...
Fetching machine types for zone 'us-east5-c'...
Fetching machine types for zone 'us-west4-a'...
Fetching machine types for zone 'southamerica-west1-b'...
Fetching machine types for zone 'us-west3-b'...
Fetching machine types for zone 'us-west4-b'...
Fetching machine types for zone 'us-west4-c'...
Exception ignored in atexit callback: <function shutdown at 0x109624940>
Exception ignored in atexit callback: <function shutdown at 0x113bf0940>
Exception ignored in atexit callback: <function shutdown at 0x115858940>
Traceback (most recent call last):
Exception ignored in atexit callback: <function reset_all at 0x10de851b0>
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/initialise.py", line 34, in reset_all
Exception ignored in atexit callback: <function shutdown at 0x104ecee60>
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/logging/__init__.py", line 2174, in shutdown
Exception ignored in atexit callback: <function reset_all at 0x10b5e51b0>
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/initialise.py", line 32, in reset_all
Exception ignored in atexit callback: <function reset_all at 0x1101b51b0>
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/initialise.py", line 34, in reset_all
    AnsiToWin32(orig_stdout).reset_all()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/ansitowin32.py", line 116, in __init__
    AnsiToWin32(orig_stdout).reset_all()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/ansitowin32.py", line 102, in __init__
    def reset_all():
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    return func(*args, **kwargs)
    return func(*args, **kwargs)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1634, in shutdown
    return func(*args, **kwargs)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1618, in shutdown
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1632, in shutdown
    have_tty = not self.stream.closed and self.stream.isatty()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/ansitowin32.py", line 59, in isatty
    self.win32_calls = self.get_win32_calls()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    return stream_isatty()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    global_worker.function_actor_manager = FunctionActorManager(global_worker)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/function_manager.py", line 72, in __init__
    sys.exit(signum)
SystemExit: 15
    for wr in reversed(handlerList[:]):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    sys.exit(signum)
    sys.exit(signum)
SystemExit: 15
SystemExit: 15
    ray._private.state.state.disconnect()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/state.py", line 51, in disconnect
    def __init__(self, worker):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    disconnect(_exiting_interpreter)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 2171, in disconnect
    sys.exit(signum)
SystemExit: 15
    def disconnect(self):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    sys.exit(signum)
SystemExit: 15
    sys.exit(signum)
    if worker.connected:
SystemExit: 15
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 453, in connected
    @property
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    sys.exit(signum)
SystemExit: 15
Fetching GPUs for zone 'us-east1-b'...
Fetching GPUs for zone 'us-east4-b'...
Fetching GPUs for zone 'us-central1-f'...
Fetching GPUs for zone 'us-west1-a'...
Fetching GPUs for zone 'europe-west1-b'...
Fetching GPUs for zone 'europe-west3-a'...
Fetching GPUs for zone 'europe-west2-a'...
Fetching GPUs for zone 'asia-southeast1-b'...
Fetching GPUs for zone 'europe-west1-d'...
Fetching GPUs for zone 'asia-east1-b'...
Fetching GPUs for zone 'europe-west3-b'...
Fetching GPUs for zone 'us-east4-a'...
Fetching GPUs for zone 'europe-west1-c'...
Fetching GPUs for zone 'europe-west2-c'...
Fetching GPUs for zone 'europe-west3-c'...
Fetching GPUs for zone 'europe-west2-b'...
Fetching GPUs for zone 'us-central1-c'...
Fetching GPUs for zone 'asia-northeast1-c'...
Fetching GPUs for zone 'asia-south1-a'...
Fetching GPUs for zone 'asia-east1-a'...
Fetching GPUs for zone 'us-east1-c'...
Fetching GPUs for zone 'us-central1-b'...
Fetching GPUs for zone 'asia-northeast1-a'...
Fetching GPUs for zone 'europe-west4-a'...
Fetching GPUs for zone 'us-central1-a'...
Fetching GPUs for zone 'europe-west4-b'...
Fetching GPUs for zone 'europe-west4-c'...
Fetching GPUs for zone 'australia-southeast1-b'...
Fetching GPUs for zone 'southamerica-east1-b'...
Fetching GPUs for zone 'us-east1-d'...
Fetching GPUs for zone 'asia-east2-b'...
Fetching GPUs for zone 'us-west1-b'...
Fetching GPUs for zone 'us-east4-c'...
Fetching GPUs for zone 'australia-southeast1-c'...
Fetching GPUs for zone 'southamerica-east1-c'...
Fetching GPUs for zone 'asia-southeast1-a'...
Fetching GPUs for zone 'asia-northeast2-c'...
Fetching GPUs for zone 'asia-east1-c'...
Fetching GPUs for zone 'asia-south1-c'...
Fetching GPUs for zone 'southamerica-east1-a'...
Fetching GPUs for zone 'us-west1-c'...
Fetching GPUs for zone 'asia-south1-b'...
Fetching GPUs for zone 'asia-south2-a'...
Fetching GPUs for zone 'asia-southeast2-b'...
Fetching GPUs for zone 'asia-east2-c'...
Fetching GPUs for zone 'australia-southeast2-c'...
Fetching GPUs for zone 'australia-southeast1-a'...
Fetching GPUs for zone 'asia-south2-b'...
Fetching GPUs for zone 'asia-southeast1-c'...
Fetching GPUs for zone 'asia-east2-a'...
Fetching GPUs for zone 'asia-northeast3-a'...
Fetching GPUs for zone 'asia-south2-c'...
Fetching GPUs for zone 'europe-north1-a'...
Fetching GPUs for zone 'europe-north1-b'...
Fetching GPUs for zone 'europe-north1-c'...
Fetching GPUs for zone 'asia-northeast2-a'...
Fetching GPUs for zone 'asia-southeast2-c'...
Fetching GPUs for zone 'asia-northeast3-b'...
Fetching GPUs for zone 'europe-southwest1-a'...
Fetching GPUs for zone 'europe-southwest1-b'...
Fetching GPUs for zone 'asia-northeast1-b'...
Fetching GPUs for zone 'europe-central2-a'...
Fetching GPUs for zone 'europe-southwest1-c'...
Fetching GPUs for zone 'europe-west12-c'...
Fetching GPUs for zone 'europe-central2-b'...
Fetching GPUs for zone 'europe-west12-a'...
Fetching GPUs for zone 'europe-west6-a'...
Fetching GPUs for zone 'europe-central2-c'...
Fetching GPUs for zone 'europe-west6-b'...
Fetching GPUs for zone 'europe-west12-b'...
Fetching GPUs for zone 'europe-west8-a'...
Fetching GPUs for zone 'europe-west6-c'...
Fetching GPUs for zone 'asia-northeast2-b'...
Fetching GPUs for zone 'europe-west9-b'...
Fetching GPUs for zone 'europe-west8-b'...
Fetching GPUs for zone 'me-central1-c'...
Fetching GPUs for zone 'asia-northeast3-c'...
Fetching GPUs for zone 'northamerica-northeast1-a'...
Fetching GPUs for zone 'northamerica-northeast2-b'...
Fetching GPUs for zone 'asia-southeast2-a'...
Fetching GPUs for zone 'australia-southeast2-a'...
Fetching GPUs for zone 'europe-west9-c'...
Fetching GPUs for zone 'europe-west8-c'...
Fetching GPUs for zone 'me-central1-a'...
Fetching GPUs for zone 'europe-west9-a'...
Fetching GPUs for zone 'northamerica-northeast1-b'...
Fetching GPUs for zone 'northamerica-northeast2-c'...
Fetching GPUs for zone 'southamerica-west1-c'...
Fetching GPUs for zone 'us-south1-a'...
Fetching GPUs for zone 'me-west1-a'...
Fetching GPUs for zone 'northamerica-northeast1-c'...
Fetching GPUs for zone 'southamerica-west1-a'...
Fetching GPUs for zone 'me-west1-b'...
Fetching GPUs for zone 'me-central1-b'...
Fetching GPUs for zone 'us-west2-b'...
Fetching GPUs for zone 'us-south1-b'...
Fetching GPUs for zone 'northamerica-northeast2-a'...
Fetching GPUs for zone 'me-west1-c'...
Fetching GPUs for zone 'australia-southeast2-b'...
Fetching GPUs for zone 'us-east5-a'...
Fetching GPUs for zone 'us-west3-c'...
Fetching GPUs for zone 'us-south1-c'...
Fetching GPUs for zone 'us-west2-c'...
Fetching GPUs for zone 'us-west3-a'...
Fetching GPUs for zone 'southamerica-west1-b'...
Fetching GPUs for zone 'us-east5-b'...
Fetching GPUs for zone 'us-west2-a'...
Fetching GPUs for zone 'us-west4-a'...
Fetching GPUs for zone 'us-west3-b'...
Fetching GPUs for zone 'us-east5-c'...
Fetching GPUs for zone 'us-west4-b'...
Fetching GPUs for zone 'us-west4-c'...
Exception ignored in atexit callback: <function shutdown at 0x107dec940>
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
Exception ignored in atexit callback: <function reset_all at 0x1082c11b0>
Exception ignored in atexit callback: <function shutdown at 0x1115a0940>
Exception ignored in atexit callback: <function shutdown at 0x107aa4940>
Traceback (most recent call last):
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/initialise.py", line 32, in reset_all
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_Exception ignored in atexit callback: <function shutdown at 0x111508940>
mode_hook.py", line 96, in wrapper
Exception ignored in atexit callback: <function reset_all at 0x10fe9d1b0>
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/thirdparty_files/colorama/initialise.py", line 32, in reset_all
    def reset_all():
    return func(*args, **kwargs)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1634, in shutdown
    return func(*args, **kwargs)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1618, in shutdown
    def reset_all():
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    @wraps(func)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    return func(*args, **kwargs)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1634, in shutdown
    disconnect(_exiting_interpreter)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 2199, in disconnect
    ray._private.state.state.disconnect()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/state.py", line 51, in disconnect
    sys.exit(signum)
SystemExit: 15
    sys.exit(signum)
SystemExit: 15
    sys.exit(signum)
SystemExit: 15
    def disconnect(self):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    ray._private.state.state.disconnect()
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/state.py", line 51, i    ray_actor._ActorClassMethodMetadata.reset_cache()
    sys.exit(signum)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/actor.py", line 228, in reset_cache
SystemExit: 15
n disconnect
    def disconnect(self):
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    @classmethod
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/ray/_private/worker.py", line 1655, in sigterm_handler
    sys.exit(signum)
SystemExit: 15
    sys.exit(signum)
SystemExit: 15
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east4-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west3-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-central1-f'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west4-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west4-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east4-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west3-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-central1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west4-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west1-d'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east1-d'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-central1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east4-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-central1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west3-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-east1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast3-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-east1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast3-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-east1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-north1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast3-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-south2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-east2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-central2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-northeast2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-north1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-southwest1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'asia-southeast2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west8-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west12-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-central2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'australia-southeast2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west9-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west8-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-north1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-southwest1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-central1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west12-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west6-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west9-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-central2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west8-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-southwest1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-west1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-west1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west12-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west6-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-central1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west9-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-west1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-west1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-south1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'europe-west6-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-south1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'me-central1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east5-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west3-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-south1-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'northamerica-northeast2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-west1-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west2-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-central2-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east5-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west2-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'southamerica-west1-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west3-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-east5-c'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west4-a'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west3-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west4-b'.
  TPU API is not enabled or you don't have TPU access to zone: 'us-west4-c'.
Traceback (most recent call last):
  File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/bilelomrani/.pyenv/versions/3.10.9/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 484, in <module>
    catalog_df = get_catalog_df(region_prefix_filter)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 450, in get_catalog_df
    tpu_df = get_tpu_df(gcp_tpu_skus)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/sky/clouds/service_catalog/data_fetchers/fetch_gcp.py", line 425, in get_tpu_df
    df['Price'] = df.apply(lambda row: get_tpu_price(row, spot=False), axis=1)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/frame.py", line 3950, in __setitem__
    self._set_item_frame_value(key, value)
  File "/Users/bilelomrani/.pyenv/versions/instruction-finetuning/lib/python3.10/site-packages/pandas/core/frame.py", line 4104, in _set_item_frame_value
    raise ValueError(
ValueError: Cannot set a DataFrame with multiple columns to the single column Price
</details>

**Comment by WoosukKwon** [comment]
I cannot reproduce the error. @bilelomrani1 Could you retry it? And could you tell us more about your environment?

**Comment by Michaelvll** [comment]
> I cannot reproduce the error. @bilelomrani1 Could you retry it? And could you tell us more about your environment?

I am guessing one possible reason is that @bilelomrani1's account may not have the access to TPUs causing the TPU dataframe empty which leads to the error.

**Comment by WoosukKwon** [comment]
> I am guessing one possible reason is that @bilelomrani1's account may not have the access to TPUs causing the TPU dataframe empty which leads to the error.

If that's the case I think we can simply merge the https://github.com/skypilot-org/skypilot-catalog/pull/24 PR and let @bilelomrani1 use the updated catalog.

**Comment by bilelomrani1** [comment]
Hi @WoosukKwon, I just tested on master with the new catalog, it works great, many thanks! Looking forward to the new release!
